Corona-Warn-App/src/main/java/de/rki/coronawarnapp/covidcertificate/pdf/ui/poster/CertificatePosterFragment.kt

```diff
@@ -74,6 +74,13 @@ class CertificatePosterFragment : Fragment(R.layout.certificate_poster_fragment)
         }
     }
 
+    override fun onDestroyView() {
+        super.onDestroyView()
+        // Delete exported pdf file
+        Timber.i("CertificatePosterFragment closed. Deleting pdf export now.")
+        viewModel.sharingIntent.value?.file?.delete()
+    }
```

Comment:
Delete can throw an exception, I would prefer if you can catch the exception just in case and delete on appScope 
so in ViewModel create 

```kotlin
    fun deleteFile() = launch(appScope) {
        try {
            sharingIntent.value?.file?.delete()
        } catch (e: Exception) {
            Timber.d(e, "deleteFile failed")
        }
    }
```

and call this method from `onDestroyView`


OR @jurajkusnier  can we use `temp`  folder I assume it should be deleted automatically

Reply:
I'm not sure whether it's really necessary to check `SecurityException` when deleting the internal file. But definitely, the implementation should be in ViewModel. It could be even called from ViewModel.onCleared() instead of from the Fragment.

Reply:
Yes. I just committed the fix, that moves the code to the ViewModel.